hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -123,6 +124,12 @@ public class HoodieWriteConfig extends HoodieConfig {
       .withDocumentation("Payload class used. Override this, if you like to roll your own merge logic, when upserting/inserting. "
           + "This will render any value set for PRECOMBINE_FIELD_OPT_VAL in-effective");
 
+  public static final ConfigProperty<String> COMBINE_ENGINE_CLASS_NAME = ConfigProperty
```

Comment:
Can we make it such that we fallback to Avro based merging (and thus support existing payloads but with a potential performance penalty) - if a engine optimized implementation does not exist. ? e.g merging using Spark's row objects

Reply:
Yeah, we provide a BWC bridge called 'HoodieAvroRecordMerge', that will be using user-defined subclass of `HoodieRecordPayload` to combine the records.

Reply:
Ack. left some comment on the migration path on the new review. resolving this